### PR TITLE
New Authorization Settings page allows users to deactivate tenants

### LINKED ISSUE #2161: Issues with Authentication with Multi-Tenant configured accounts
I am unable to authenticate into Azure Batch Explorer.  My Azure AD account is linked as a guest account to multiple Microsoft Tenants, due to the nature of working in the service provider field.  Several of these clients have either marked my guest account as disabled, or have advanced conditional access policy configurations that are preventing authentication of my account.  

Once my authentication hits only one of these client tenants, I am unable to gain access to using Batch Explorer.  If I try to cancel out of the authentication, Batch Explorer closes immediately.

I am requesting either a bug fix or a feature enhancement.
- Bug Fix:  Error handling to be added to work with authentication failure of multiple tenant users
- Feature Request:  Enhanced AAD Tenant selection capabilities after initial authentication

### PR DESCRIPTION
**Pull request by gingi**
Allows users to select which tenant to authenticate against. When a user's AAD credentials are associated with multiple tenants, a failed tenant will not prevent users from using BatchExplorer on authorized tenants. When a non-home tenant authorization fails, a notification is displayed asking the user to configure their authentication settings. The authentication settings page displays a list of associated tenants and allows users to deactivate any and to reauthenticate newly activated ones.

Includes various other minor updates.

Fixes #2161 

# Testing

## User types:
* User with a single tenant
* User with multiple valid tenants
* User with multiple tenants, some invalid

## All User Types
* Start the app as a logged in user
* Log out and back in
* Log out, quit, then start the app

## Multiple-Tenant Users
* As a user with multiple tenants, disable a non-home tenant, refresh subscriptions
* Log out, restart the app, ensure tenant is still disabled

## Users with Multiple Tenants, Some Invalid
* Ensure a tenant error notification is shown initially
* Disable the tenant, refresh subscriptions → there should be no error
* Reenable the invalid tenant; should automatically reauthenticate

Pasted_Image_10_15_21__4_24_PM 

Pasted_Image_10_15_21__4_44_PM

### CODE REVIEW

**Comment by gingi** [comment]
I would like to add a few more unit tests.

File: src/@batch-flask/core/aad/access-token-cache/access-token-cache.spec.ts [hunk 1]
```diff
@@ -5,13 +5,12 @@ import { AccessTokenCache } from "./access-token-cache";
 const tenant1 = "tenant-1";
 const resource1 = "http://example.com";
 const token1 = {
-    access_token: "sometoken",
-    expires_on: DateTime.local().plus({ hours: 1 }).toJSDate(),
+    accessToken: "sometoken",
```
Comment (cRui861) [inline src/@batch-flask/core/aad/access-token-cache/access-token-cache.spec.ts:8]: Nit: Why are some of these updated to camel case but others are left as snake case?

**Comment by themorey** [comment]
I built a Mac client from this feature branch to get around a dead tenant on my account.  It was very helpful, thanks!

**Review by dpwatrous — APPROVED** [review]

**Comment by dpwatrous** [comment]
Need a rebase on master if we want to preserve commits, but otherwise looks good to me!

**Pull request merged**